### ISSUE TITLE
Release v2.3

### ISSUE THREAD

**Issue by galtm** [xspec/xspec#1657]
I would like to release v2.3 before the end of 2022, maybe in October. I might need community help with some of the TODOs listed below, especially since I haven't done a release before.

(cc'ing @xspec/xspec, @xspec/maven-plugin, @xspec/oxygen-xml-editor-xspec-support, @vincentml)

---

### TODOs

(based on [Release Workflow](https://github.com/xspec/xspec/wiki/Release-Workflow))

1. Prepare the release

   - [ ] Release `2.3` to Oxygen add-on channel (Stable Release Candidate)

   - [ ] Notify the XSpec discussion list of the upcoming `v2.3` release.

     - Notable changes TBD

   - [ ] Test the add-on `2.3` with Oxygen 24.1
     - All the 4 transformation scenarios should work. (Don't forget to test **Run XSpec Test** with XSLT, XQuery and Schematron.)
     - Loading `xspec.xpr` should disable the add-on and enable its own framework.

   - [ ] Draft a new release (`v2.3`) at [Releases page](https://github.com/xspec/xspec/releases)
   - [ ] Create a pull request incrementing the version (to `2.3` without `-SNAPSHOT`) of `pom.xml` and the Oxygen add-on and merge it into `master`

1. Release it

   - [ ] Make sure `2.3` has been deployed to Maven Central successfully (automated)
   - [ ] Publish the release at [Releases page](https://github.com/xspec/xspec/releases)
   - Promote the release `v2.3`
     - [ ] XSpec discussion list
     - [ ] xml.com
     - [ ] xml mailing list
     - [ ] tweet
     - [ ] Write post on [`xspec.io`](https://github.com/xspec/xspec.github.io)

1. Update Wiki
   - [ ] Complete [the documentation issues](https://github.com/xspec/xspec/issues?q=is%3Aissue+is%3Aopen+label%3Adocumentation+milestone%3Av2.3)

1. Create a pull request for [the Oxygen XSpec Helper View](https://github.com/xspec/oXygen-XML-editor-xspec-support)
   - [ ] Create it

1. Prepare the next development

   - [ ] Close the project and the milestone related to `v2.3`
   - [ ] Create the project and the milestone for `v2.4`

1. Start the next development
   - [ ] Create a pull request incrementing the version (to `2.4-SNAPSHOT`) of `pom.xml` and merge it into master

   - (If something critical is found, we will create a branch for `2.3.1-SNAPSHOT` and work on it.)

**Comment by galtm** [comment]
@AirQuick, how did you generate v2.2 release notes in the table format? I drafted release notes with a hand-crafted summary and generated data following, but the generated data is a long list. Did you start from the list and manually reorganize the items to form the tables?

@AirQuick and @cirulls, my draft is on the [Releases](https://github.com/xspec/xspec/releases) page. Any feedback is welcome!

**Comment by AirQuick** [comment]
Thanks @galtm for moving this project forward. As mention in https://github.com/xspec/xspec/issues/601#issuecomment-619644748, the relnote generator used for v2.2 is a simple XSLT stylesheet which unfortunately depends on my closed library. I'll run it for v2.3 and update the draft if necessary.

**Comment by galtm** [comment]
@AirQuick , @cirulls : Can we move toward releasing v2.3?

@AirQuick , thank you for offering to run your relnote generator that has a private dependency. When might you be available to do that? If you're too busy in the next several weeks, I can use the standard GitHub generated release notes (although your relnote generator is better).

**Comment by AirQuick** [comment]
Sorry for keeping you too long. #1761 will be the last PR before starting the release workflow.

**Comment by AirQuick** [comment]
Updated the draft release note. "What's Changed" section has been replaced with auto-generated "Merged Pull Requests" section and "Closed Issues" section.

**Comment by galtm** [comment]
Thanks! I was about to say that the item I wrote awhile back about gaps in Saxon 11 support is stale, but then I refreshed the page and saw that you have already updated that.  : ) What I see now looks great.

**Comment by AirQuick** [comment]
A few more release note updates:

* Replaced these items in "Common to Languages Under Test" section

  > * Saxon 10 is recommended.
  > * Saxon 11 is compatible with some XSpec functionality, but there are some gaps (e.g., running XSpec with XProc under Saxon 11 is not tested yet).

  with

  > * XSpec is now tested on Saxon 12, 11, 10, and 9.9, while Saxon 9.9 is deprecated.

  to reflect the latest status.

* Removed this item from "Common to Languages Under Test" section:

  > * Log4j uses 2.17.1, which includes security fixes.

  because Log4j has been removed from the test.

* Removed this item from "Common to Languages Under Test":

  > * Command line support fixes a bug that interfered with MSYS on Windows.

  because there are no tests for MSYS and I'm afraid highlighting it is a bit misleading. (Of course, the fix itself is useful and welcome.)

* Removed this section. While this list looks very nice and I like ti, I'm not sure we can maintain it in the current form:

  > ## New Contributors
  > * `@mcwayliffe` made their first contribution in https://github.com/xspec/xspec/pull/1568
  > * `@danielnaab` made their first contribution in https://github.com/xspec/xspec/pull/1620
  > * `@ndw` made their first contribution in https://github.com/xspec/xspec/pull/1610

**Comment by galtm** [comment]
> * I'm not sure we can maintain it in the current form

GitHub generated that content automatically when I first created the v2.3 release notes draft. Shouldn't the GitHub behavior make maintenance easy on our side? If your typical process bypasses the automatically generated GitHub release notes and, therefore, creating the list of new contributors would be an extra step, then I'm fine with leaving it out to simplify the release process.

**Comment by galtm** [comment]
I didn't notify the XSpec discussion list about the upcoming release. I confess to some reluctance because it's a G..gle list. As the list has had no traffic in 9 months, maybe it has low readership and skipping the early notification step is not that bad.

Post-release, is there someone else who can do the notifications to the XSpec discussion list and Twitter? (Maybe @cirulls or @cmarchand ?)

**Comment by cmarchand** [comment]
Release 2.3, 2.3.1 and 2.3.2 are not available on [maven central](https://repo1.maven.org/maven2/io/xspec/xspec/).

Previous releases were uploaded by build process.

Is this normal ?

**Comment by galtm** [comment]
I don't see any snapshots or release candidates at https://repo1.maven.org/maven2/io/xspec/xspec/ , only final releases. I haven't merged #1781 onto the master branch yet, so not seeing v2.3.2 on Maven Central yet is normal.

The Maven Deploy log from #1773 has one (nonfatal) warning, below. It doesn't seem to prevent deployment. Besides, that file predates v2.2.4, and replacing `-SNAPSHOT` by `.1` makes the path shorter than 100 characters.
```
2023-08-28T09:44:04.7423700Z [INFO] Reading assembly descriptor: assembly/enduser-files.xml
2023-08-28T09:44:04.8575565Z [INFO] Building tar: /home/runner/work/xspec/xspec/target/xspec-2.3-SNAPSHOT-enduser-files.tar.gz
2023-08-28T09:44:04.9282715Z [WARNING] Entry: xspec-2.3-SNAPSHOT/src/compiler/xquery/invoke-compiled/invoke-compiled-current-scenario-or-expect.xsl longer than 100 characters.
2023-08-28T09:44:04.9284680Z [WARNING] Resulting tar file can only be processed successfully by GNU compatible tar commands
2023-08-28T09:44:05.0329520Z [INFO] Building zip: /home/runner/work/xspec/xspec/target/xspec-2.3-SNAPSHOT-enduser-files.zip
```

I'm not an expert in Maven, so I might be missing something. Whenever I merge #1781, I'll definitely keep my eyes open for the deployment.

**Comment by AirQuick** [comment]
> > * I'm not sure we can maintain it in the current form
> 
> GitHub generated that content automatically when I first created the v2.3 release notes draft. Shouldn't the GitHub behavior make maintenance easy on our side? If your typical process bypasses the automatically generated GitHub release notes and, therefore, creating the list of new contributors would be an extra step, then I'm fine with leaving it out to simplify the release process.

At the time of that writing, I wasn't sure how the `Generate release notes` button would work with the existing draft. Today I tried the button and found that it just appended its auto content to the latest draft.
I've put the **New Contributors** subsection into the **Merged Pull Requests** section.

**Comment by AirQuick** [comment]
> I didn't notify the XSpec discussion list about the upcoming release.

I've posted a very short [message](https://groups.google.com/g/xspec-users/c/kzFFl953bwI/m/GLtWnyBLAgAJ) to the list. I'll post a post-release message too.

**Comment by galtm** [comment]
@AirQuick : Thank you for your help with the discussion list and Release Notes! The New Contributors section looks good.

**Comment by galtm** [comment]
The release and its release notes have been published. Artifacts are available on Maven Central.

The announcement on the xml-dev mailing list is visible here: http://lists.xml.org/archives/xml-dev/202310/msg00000.html

I submitted a post on xml.com and will check tomorrow to see if it appears.

I drafted a [post for xspec.io](https://github.com/xspec/xspec.github.io/pull/14), but I'm having some trouble that I will work on another day.

**Comment by AirQuick** [comment]
> Promote the release `v2.3`
> * [x]  XSpec discussion list

https://groups.google.com/g/xspec-users/c/bn8o2ZtG4Zk

**Comment by AirQuick** [comment]
> Create a pull request for [the Oxygen XSpec Helper View](https://github.com/xspec/oXygen-XML-editor-xspec-support)

Done: xspec/oXygen-XML-editor-xspec-support#51

**Comment by AirQuick** [comment]
> * [ ]  Create the project and the milestone for `v3.0`

The projects have not been used actively. I think we can skip creating the project and just use the milestone.

**Comment by galtm** [comment]
> > * [ ]   Create the project and the milestone for `v3.0`
> 
> The projects have not been used actively. I think we can skip creating the project and just use the milestone.

OK. Thanks for creating a milestone for v3.0. I closed the project and milestone for v2.3.

**Comment by galtm** [comment]
> I submitted a post on xml.com and will check tomorrow to see if it appears.

It is live: https://www.xml.com/news/2023-10-xspec-v232/

**Comment by galtm** [comment]
> * [ ]  Create a pull request incrementing the version (to `3.0-SNAPSHOT`) of `pom.xml` and merge it into master

Created https://github.com/xspec/xspec/pull/1784

Before merging it, I'll wait for checks to finish, look at logs to see if everything looks OK, and wait several days for a code review. The change is formulaic enough that I won't wait a really long time, though.

After merging it, I'll look at the backlog of pull requests and merge the ones that seem ready.

**Comment by galtm** [comment]
> Done: [xspec/oXygen-XML-editor-xspec-support#51](https://github.com/xspec/oXygen-XML-editor-xspec-support/pull/51)

Thank you! I started working on that but I'm grateful that you stepped in because (a) I wasn't sure how to test my changes in Oxygen before the PR would have been merged, and (b) I'm not familiar with the Java code or how to run the tests in that repo.

**Comment by galtm** [comment]
> I drafted a [post for xspec.io](https://github.com/xspec/xspec.github.io/pull/14), but I'm having some trouble that I will work on another day.

Did a local build and attached screen capture to https://github.com/xspec/xspec.github.io/pull/14

**Comment by galtm** [comment]
> As you ask for a Twitter communication, I though the release was published.

@cmarchand : Now that the release is published, would you be willing to do a Twitter communication when you get a chance? Thank you!

**Comment by galtm** [comment]
Thanks for the tweet, @cmarchand ! I hadn't realized that you had already done it. : ) 

> I was reluctant to thank you in the tweet without your authorization... could I do next time ?

What you wrote was fine. It is also fine not to single me out.

**Comment by galtm** [comment]
All the checklist actions on our side are done, but I'll keep this issue open until the xspec.io PR gets merged.

> > I drafted a [post for xspec.io](https://github.com/xspec/xspec.github.io/pull/14), but I'm having some trouble that I will work on another day.
> 
> Did a local build and attached screen capture to [xspec/xspec.github.io#14](https://github.com/xspec/xspec.github.io/pull/14)

**Comment by AirQuick** [comment]
Closing as completed :smiley:

**Comment by galtm** [comment]
The post for xspec.io is now showing up at https://xspec.io/posts/xspec_232_release/ . 

To everyone who helped work on and release v2.3, thank you!!

<html>
<body>


:blossom:
--



</body>
</html>
